flatland_server/include/flatland_server/message_system/message_server.h

```diff
@@ -0,0 +1,50 @@
+#ifndef FLATLAND_SERVER_MESSAGE_SERVER_H
+#define FLATLAND_SERVER_MESSAGE_SERVER_H
+
+#include <map>
+#include <vector>
+#include <queue>
+
+namespace flatland_server {
+
+    class MessageServer;
+    template <class T>
+    class Subscriber;
+    template <class T>
+    class Publisher;
+    class MessageTopicBase {};
+
+    enum MessageType {
+        UIAction,
+        HumanAction
+    };
+
+    template <class T>
+    class MessageTopic : public MessageTopicBase {
+    public:
+        MessageType type;
+        std::string source;
+        std::queue<T> *messages;
```

Comment:
Please use consistent C++ style pointer type alignment.
Also, why is this a pointer to begin with?